editor/src/extension.js

```diff
@@ -321,18 +321,24 @@ const Exporter = () => {
 				_ti_tpc_screenshot_url,
 				! _ti_tpc_published,
 				link
-			).then( ( r ) => {
+			).then( async ( r ) => {
 				if ( r.success ) {
-					setPublished( ! published );
-					saveMeta();
-					createSuccessNotice(
-						published
-							? __( 'Template Unpublished.', 'templates-patterns-collection' )
-							: __( 'Template Published.', 'templates-patterns-collection' ),
-						{
-							type: 'snackbar',
+					await getTemplate( _ti_tpc_template_id ).then( ( results ) => {
+						if ( _ti_tpc_template_id === results.template_id ) {
+							setScreenshotURL( results.template_thumbnail );
+							setPublished( ! published );
+							saveMeta();
+							createSuccessNotice(
+								published
+									? __( 'Template Unpublished.', 'templates-patterns-collection' )
+									: __( 'Template Published.', 'templates-patterns-collection' ),
+								{
+									type: 'snackbar',
+								}
+							);
 						}
-					);
+					} );
```

Comment:
We can add a catch to this to ensure that stuff still works if the request fails. We should probably do the same for the `publishTemplate` call above.